pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_partial_stats failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_partial_stats [failed](https://tanzanite.cluster.engflow.com/invocations/default/fa2428d1-bdd9-49bb-81b2-624f58c79ea7?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4ZWNidWlsZGVyL3Rlc3RzL2xvY2FsOmxvY2FsX3Rlc3Q=) on master @ [8556927831d2cce730a64d8a5a7a0f660a57b060](https://github.com/cockroachdb/cockroach/commits/8556927831d2cce730a64d8a5a7a0f660a57b060):

Fatal error:

```
panic: test timed out after 1h16m40s
running tests:
	TestExecBuild_partial_stats (1h16m26s)

```
Stack: 

```
goroutine 608600 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestExecBuild_partial_stats
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestExecBuild_partial_stats646433795
    test_log_scope.go:76: use -show-logs to present logs inline
    logic.go:3095: let $forecastsEnabledPrev = false
[11:18:51] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/partial_stats: 23 statements
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_partial_stats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45959

```
[11:18:51] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/partial_stats: 23 statements
panic: test timed out after 1h16m40s
```

So we've made it at least as far as the `ALTER TABLE ab DROP COLUMN a`.

Here's the interesting section of logs where we stop making progress, which shows that we make it to `CREATE TABLE ka (k INT PRIMARY KEY, a INT, INDEX(a))`:

```
I250103 11:18:48.177198 9565 sql/schemachanger/scbuild/event_log.go:61 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 236  event logging for declarative schema change targets built for ALTER TABLE
I250103 11:18:48.178316 9565 sql/schemachanger/scplan/plan.go:95 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 237  building declarative schema changer plan in StatementPhase (rollback=false) for ALTER TABLE
I250103 11:18:48.326612 9565 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 238  executing declarative schema change StatementPhase stage 1 of 1 with 9 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:48.330461 9565 sql/schemachanger/scplan/plan.go:95 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 239  building declarative schema changer plan in PreCommitPhase (rollback=false) for ALTER TABLE
I250103 11:18:48.461614 9565 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 240  executing declarative schema change PreCommitPhase stage 1 of 2 with 1 MutationType op (rollback=false) for ALTER TABLE
I250103 11:18:48.461882 9565 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 241  executing declarative schema change PreCommitPhase stage 2 of 2 with 13 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:48.472724 9565 sql/index_split_scatter.go:126 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 242  making a single split point in tableId=109 index=3
I250103 11:18:48.516456 9565 sql/index_split_scatter.go:126 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 243  making a single split point in tableId=109 index=4
I250103 11:18:48.615151 9565 sql/conn_executor_ddl.go:113 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 244  queued new schema change job 1034920684174901249 using the new schema changer
I250103 11:18:48.630680 9565 5@util/log/event_log.go:39 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 245 ={"Timestamp":1735903128088113736,"EventType":"alter_table","Statement":"ALTER TABLE ‹test›.public.‹ab› DROP COLUMN ‹a›","Tag":"ALTER TABLE","User":"root","DescriptorID":109,"TableName":"‹test.public.ab›","MutationID":1}
I250103 11:18:48.632224 9565 jobs/wait.go:56 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 246  waiting for 1 [1034920684174901249] queued jobs to complete
I250103 11:18:48.632568 11044 jobs/adopt.go:224 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root,job=1034920684174901249] 247  job 1034920684174901249: resuming execution
I250103 11:18:48.665063 11052 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=1034920684174901249] 248  NEW SCHEMA CHANGE job 1034920684174901249: stepping through state running
I250103 11:18:48.779610 11052 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=1034920684174901249] 249 ={"Timestamp":1735903128779592279,"EventType":"status_change","JobID":1034920684174901249,"JobType":"NEW SCHEMA CHANGE","Description":"status changed to: running","PreviousStatus":"running","NewStatus":"running","RunNum":0}
I250103 11:18:48.813389 11052 sql/schemachanger/scrun/scrun.go:315 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 250  rebuilding declarative schema change state from descriptors [109]
I250103 11:18:48.814790 11052 sql/schemachanger/scplan/plan.go:95 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 251  building declarative schema changer plan in PostCommitPhase (rollback=false) for ALTER TABLE
I250103 11:18:48.959947 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 252  executing declarative schema change PostCommitPhase stage 1 of 7 with 3 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:49.124860 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 253  executing declarative schema change PostCommitPhase stage 2 of 7 with 1 BackfillType op (rollback=false) for ALTER TABLE
I250103 11:18:49.198843 7403 kv/kvserver/queue.go:616 ⋮ [T1,Vsystem,n1,s1] 254  rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I250103 11:18:49.570783 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 255  executing declarative schema change PostCommitPhase stage 3 of 7 with 3 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:49.746412 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 256  executing declarative schema change PostCommitPhase stage 4 of 7 with 3 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:49.924289 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 257  executing declarative schema change PostCommitPhase stage 5 of 7 with 1 BackfillType op (rollback=false) for ALTER TABLE
I250103 11:18:50.096450 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 258  executing declarative schema change PostCommitPhase stage 6 of 7 with 4 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:50.270605 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 259  executing declarative schema change PostCommitPhase stage 7 of 7 with 1 ValidationType op (rollback=false) for ALTER TABLE
I250103 11:18:50.304227 12304 sql/backfill.go:1940 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 261  validation: table ‹ab› row count = 0, took 22.996688ms
I250103 11:18:50.304149 12303 sql/backfill.go:1894 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 260  validation: index ‹ab›/‹crdb_internal_index_3_name_placeholder› row count = 0, time so far 22.872388ms
I250103 11:18:50.472067 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 262  executing declarative schema change PostCommitNonRevertiblePhase stage 1 of 3 with 15 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:50.675229 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 263  executing declarative schema change PostCommitNonRevertiblePhase stage 2 of 3 with 7 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:50.852492 11052 sql/schemachanger/scrun/scrun.go:186 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 264  executing declarative schema change PostCommitNonRevertiblePhase stage 3 of 3 with 7 MutationType ops (rollback=false) for ALTER TABLE
I250103 11:18:51.060881 11052 5@util/log/event_log.go:39 ⋮ [T1,Vsystem,n1,job=NEW SCHEMA CHANGE id=1034920684174901249] 265 ={"Timestamp":1735903130851666827,"EventType":"finish_schema_change","InstanceID":1,"DescriptorID":109,"LatencyNanos":2362961384}
I250103 11:18:51.081393 11052 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=1034920684174901249] 266  NEW SCHEMA CHANGE job 1034920684174901249: stepping through state succeeded
I250103 11:18:51.081781 12646 jobs/adopt.go:224 ⋮ [T1,Vsystem,n1,job=1034920692500594689] 267  job 1034920692500594689: resuming execution
I250103 11:18:51.105411 12658 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=1034920692500594689] 268  SCHEMA CHANGE GC job 1034920692500594689: stepping through state running
I250103 11:18:51.198458 12658 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=1034920692500594689] 269 ={"Timestamp":1735903131198442743,"EventType":"status_change","JobID":1034920692500594689,"JobType":"SCHEMA CHANGE GC","Description":"status changed to: running","PreviousStatus":"running","NewStatus":"running","RunNum":0}
I250103 11:18:51.265284 11052 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=1034920684174901249] 270 ={"Timestamp":1735903131265265938,"EventType":"status_change","JobID":1034920684174901249,"JobType":"NEW SCHEMA CHANGE","Description":"status changed to: succeeded","PreviousStatus":"running","NewStatus":"succeeded","RunNum":0}
I250103 11:18:51.305912 9565 jobs/wait.go:148 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 271  waited for 1 [1034920684174901249] queued jobs to complete 2.673436117s
I250103 11:18:51.308205 7106 sql/logictest/logic.go:1203 ⋮ [-] 272  --- progress: ‹/var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/partial_stats›: 23 statements
I250103 11:18:51.399141 9565 1@util/log/event_log.go:39 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 273 ={"Timestamp":1735903131384005670,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"sql.stats.forecasts.enabled\" = false","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"sql.stats.forecasts.enabled","Value":"‹false›"}
I250103 11:18:51.400501 7106 sql/logictest/logic.go:3488 ⋮ [-] 274  Rewrote test statement:
I250103 11:18:51.400501 7106 sql/logictest/logic.go:3488 ⋮ [-] 274 +‹CREATE TABLE ka (k INT8 PRIMARY KEY, a INT8, INDEX (a), FAMILY (k), FAMILY (a));›
I250103 11:18:51.422638 12658 sql/gcjob/gc_job_utils.go:273 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 275  updated progress status: waiting for GC TTL, payload: ‹indexes:<index_id:1 > indexes:<index_id:2 > indexes:<index_id:4 > ranges_unsplit_done:true ›
I250103 11:18:51.429010 9565 5@util/log/event_log.go:39 ⋮ [T1,Vsystem,n1,client=127.0.0.1:52424,hostssl,user=root] 276 ={"Timestamp":1735903131401314589,"EventType":"create_table","Statement":"CREATE TABLE ‹test›.public.‹ka› (‹k› INT8 PRIMARY KEY, ‹a› INT8, INDEX (‹a›), FAMILY (‹k›), FAMILY (‹a›))","Tag":"CREATE TABLE","User":"root","DescriptorID":110,"TableName":"‹test.public.ka›"}
I250103 11:18:51.546773 12658 sql/gcjob/gc_job_utils.go:273 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 277  updated progress status: deleting data, payload: ‹indexes:<index_id:1 > indexes:<index_id:2 > indexes:<index_id:4 > ranges_unsplit_done:true ›
I250103 11:18:51.547127 12658 sql/schema_changer.go:1467 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 278  waiting for a single version...
I250103 11:18:51.563255 12658 sql/schema_changer.go:1472 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 279  waiting for a single version... done (at v 9), took 16.124239ms
I250103 11:18:51.563374 12658 sql/gcjob/index_garbage_collection.go:165 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 280  clearing index 1 from table 109
I250103 11:18:51.565752 12658 sql/gcjob/gc_job_utils.go:53 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 281  marked index 1 as GC'd
I250103 11:18:51.565868 12658 sql/gcjob/index_garbage_collection.go:165 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 282  clearing index 2 from table 109
I250103 11:18:51.568476 12658 sql/gcjob/gc_job_utils.go:53 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 283  marked index 2 as GC'd
I250103 11:18:51.568569 12658 sql/gcjob/index_garbage_collection.go:165 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 284  clearing index 4 from table 109
I250103 11:18:51.571124 12658 sql/gcjob/gc_job_utils.go:53 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 285  marked index 4 as GC'd
I250103 11:18:51.665686 12658 sql/gcjob/gc_job_utils.go:273 ⋮ [T1,Vsystem,n1,job=SCHEMA CHANGE GC id=1034920692500594689] 286  updated progress status: waiting for MVCC GC, payload: ‹indexes:<index_id:1 status:WAITING_FOR_MVCC_GC > indexes:<index_id:2 status:WAITING_FOR_MVCC_GC > indexes:<index_id:4 status:WAITING_FOR_MVCC_GC > ranges_unsplit_done:true ›
W250103 11:19:08.971378 13516 kv/kvserver/spanlatch/manager.go:606 ⋮ [T1,Vsystem,n1,s1,r76/1:‹/{Table/109/4-Max}›] 287  have been waiting 15s to acquire write latch ‹/Local/Range/Table/110/1/3484/0/Transaction/"6b5856d4-af4d-4025-96c1-45c96601aa4a"›@0,0 for request HeartbeatTxn [/Table/110/1/‹3484›/‹0›], [txn: 6b5856d4], held by write latch ‹/Local/Range/Table/110/1/3484/0/Transaction/"6b5856d4-af4d-4025-96c1-45c96601aa4a"›>
I250103 11:19:14.002236 7820 jobs/adopt.go:105 ⋮ [T1,Vsystem,n1] 288  claimed 5 jobs
I250103 11:19:14.049154 16938 jobs/adopt.go:224 ⋮ [T1,Vsystem,n1,job=101] 289  job 101: resuming execution
I250103 11:19:14.051505 16936 jobs/adopt.go:224 ⋮ [T1,Vsystem,n1,job=105] 290  job 105: resuming execution
I250103 11:19:14.053678 16937 jobs/adopt.go:224 ⋮ [T1,Vsystem,n1,job=100] 291  job 100: resuming execution
I250103 11:19:14.055757 16939 jobs/adopt.go:224 ⋮ [T1,Vsystem,n1,job=103] 292  job 103: resuming execution
I250103 11:19:14.057839 16940 jobs/adopt.go:224 ⋮ [T1,Vsystem,n1,job=104] 293  job 104: resuming execution
I250103 11:19:14.128323 16953 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=105] 294  UPDATE TABLE METADATA CACHE job 105: stepping through state running
I250103 11:19:14.165985 16968 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=101] 295  POLL JOBS STATS job 101: stepping through state running
I250103 11:19:14.175339 16978 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=103] 296  AUTO UPDATE SQL ACTIVITY job 103: stepping through state running
I250103 11:19:14.192663 16973 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=104] 297  MVCC STATISTICS UPDATE job 104: stepping through state running
I250103 11:19:14.255144 16994 jobs/registry.go:1679 ⋮ [T1,Vsystem,n1,job=100] 298  KEY VISUALIZER job 100: stepping through state running
I250103 11:19:14.557682 16968 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=101] 299 ={"Timestamp":1735903154557666191,"EventType":"status_change","JobID":101,"JobType":"POLL JOBS STATS","Description":"status changed to: running","PreviousStatus":"running","NewStatus":"running","RunNum":0}
I250103 11:19:14.568972 16953 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=105] 300 ={"Timestamp":1735903154568956299,"EventType":"status_change","JobID":105,"JobType":"UPDATE TABLE METADATA CACHE","Description":"status changed to: running","PreviousStatus":"running","NewStatus":"running","RunNum":0}
I250103 11:19:14.605265 16978 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=103] 301 ={"Timestamp":1735903154605248007,"EventType":"status_change","JobID":103,"JobType":"AUTO UPDATE SQL ACTIVITY","Description":"status changed to: running","PreviousStatus":"running","NewStatus":"running","RunNum":0}
I250103 11:19:14.605665 16978 sql/sql_activity_update_job.go:79 ⋮ [T1,Vsystem,n1,job=AUTO UPDATE SQL ACTIVITY id=103] 302  starting sql stats activity flush job
I250103 11:19:14.655375 16973 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=104] 303 ={"Timestamp":1735903154655364794,"EventType":"status_change","JobID":104,"JobType":"MVCC STATISTICS UPDATE","Description":"status changed to: running","PreviousStatus":"running","NewStatus":"running","RunNum":0}
I250103 11:19:14.655594 16994 1@jobs/structured_log.go:48 ⋮ [T1,Vsystem,n1,job=100] 304 ={"Timestamp":1735903154655585154,"EventType":"status_change","JobID":100,"JobType":"KEY VISUALIZER","Description":"status changed to: running","PreviousStatus":"running","NewStatus":"running","RunNum":0}
I250103 11:19:14.655649 16973 sql/mvcc_statistics_update_job.go:63 ⋮ [T1,Vsystem,n1,job=MVCC STATISTICS UPDATE id=104] 305  starting mvcc statistics update job
```

To me, it looks like we've finished the `CREATE TABLE ka (k INT PRIMARY KEY, a INT, INDEX(a))` and we're now stuck or very slowly executing `INSERT INTO ka select x, x from generate_series(0, 9998) as g(x)`.

--- a/pkg/sql/opt/exec/execbuilder/testdata/partial_stats
+++ b/pkg/sql/opt/exec/execbuilder/testdata/partial_stats
@@ -1,4 +1,4 @@
-# LogicTest: local
+# LogicTest: local !metamorphic-batch-sizes
 
 # Tests that verify we can create and use partial table statistics
 # USING EXTREMES correctly.

